modules/core/src/test/java/org/apache/ignite/internal/processors/affinity/GridAffinityAssignmentV2Test.java

```diff
@@ -267,4 +265,66 @@ protected TcpDiscoveryNode node(DiscoveryMetricsProvider metrics, IgniteProductV
 
         return node;
     }
+
+    /**
+     * Test effect change ignite system property.
+     */
+    @Test
+    public void testChangeIgniteSystemPropertiesForAffinityAssigment() {
```

Comment:
This test depends on system properties of the runnning machine. Refactor it by setting properties on test start and change them during the test to cover all cases.